pandas/io/excel/_openpyxl.py

```diff
@@ -525,8 +525,28 @@ def _convert_cell(self, cell, convert_float: bool) -> Scalar:
         return cell.value
 
     def get_sheet_data(self, sheet, convert_float: bool) -> List[List[Scalar]]:
+        # GH 39001
+        # Reading of excel file depends on dimension data being correct but
+        # writers sometimes omit or get it wrong
+        sheet.reset_dimensions()
+
         data: List[List[Scalar]] = []
-        for row in sheet.rows:
-            data.append([self._convert_cell(cell, convert_float) for cell in row])
+        last_row_with_data = -1
+        for row_number, row in enumerate(sheet.rows):
+            converted_row = [self._convert_cell(cell, convert_float) for cell in row]
+            if not all(cell == "" for cell in converted_row):
+                last_row_with_data = row_number
+            data.append(converted_row)
+
+        if len(data) > 0:
+            # Trim trailing empty rows
+            data = data[: last_row_with_data + 1]
+
+            # With dimension reset, openpyxl no longer pads rows
+            max_width = max(len(data_row) for data_row in data)
```

Comment:
With some xlsx files I get an error:

`max() arg is an empty sequence`

I am pretty sure this happens when data object is empty

Reply:
I can't share the file but I can tell it starts with empty rows and empty columns.

Reply:
Yep - thanks for catching this!